pkg/controller/elasticsearch/elasticsearch_controller.go

```diff
@@ -307,14 +307,13 @@ func (r *ReconcileElasticsearch) updateStatus(
 	return r.Status().Update(cluster)
 }
 
-// finalizersFor returns the list of finalizers applying to a given es cluster
-func (r *ReconcileElasticsearch) finalizersFor(
-	es elasticsearchv1beta1.Elasticsearch,
-) []finalizer.Finalizer {
-	clusterName := k8s.ExtractNamespacedName(&es)
-	return []finalizer.Finalizer{
-		r.esObservers.Finalizer(clusterName),
-		keystore.Finalizer(k8s.ExtractNamespacedName(&es), r.dynamicWatches, es.Kind),
-		http.DynamicWatchesFinalizer(r.dynamicWatches, es.Kind, es.Name, elasticsearchv1beta1.ESNamer),
-	}
+// onDelete garbage collect resources when a Elasticsearch cluster is deleted
+func (r *ReconcileElasticsearch) onDelete(
+	es types.NamespacedName,
+) {
+	// Stop tracking that cluster in expectations.
```

Comment:
I would either remove that comment (`expectations.RemoveCluster ` feels self-explanatory), or add comments about the other stuff we do here (observers & dynamic watches).